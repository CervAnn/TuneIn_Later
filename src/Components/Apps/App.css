* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Permanent Marker', cursive;
}

header {
  height: 175px;
  width: 100%;
  font-size: 3em;
  background-color: #596743;
  color: #F4E697;
}

img {
  height: 45px;
  width: 5%;
  margin-left: 10px;
}

.headerContents {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.buttonContainer {
text-align: center;
}

#newUser-button, #existingUser-button {
  font-size: 20px;
  color: #F4E697;
  background-color: transparent;
  background-position: center;
  border: none;
  margin-right: 30px;
}

#newUser, #existingUser {
  height: auto;
  width: 3%;
  margin-right: 8px;
}

body {
  background-color: #384150
}


