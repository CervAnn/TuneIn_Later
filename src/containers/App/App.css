* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: "Permanent Marker", cursive;
}

header {
  height: 225px;
  width: 100%;
  font-size: 3em;
  background-color: #596743;
  color: #f4e697;
}

#headphones {
  margin-left: 10px;
}

a {
  text-decoration: none;
  color: #f4e697;
}

img {
  height: 45px;
  width: 5%;
}

.headerContents {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.buttonContainer {
  text-align: center;
  display: flex;
  justify-content: center;
  margin: auto;
}

#newUser-button,
#existingUser-button,
#logoutUser-button,
#viewFavorites-button {
  font-size: 20px;
  color: #f4e697;
  background-color: transparent;
  background-position: center;
  border: none;
  margin-right: 30px;
}

#newUser,
#existingUser,
#logoutUser,
#viewFavorites {
  height: auto;
  width: 3%;
  margin-right: 8px;
}

#newUser-button:hover,
#existingUser-button:hover,
#logoutUser-button:hover,
#viewFavorites-button:hover,
h1:hover {
  border: 2px solid #cc7b3a;
  border-radius: 15px;
}

body {
  background-color: #384150;
}
