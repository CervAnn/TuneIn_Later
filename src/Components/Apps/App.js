import React, { Component }  from 'react';
import './App.css';
import SearchForm from '../Forms/SearchForm'
import Container from '../Container/Container';
import { Router, Route, Link, Redirect } from 'react-router-dom';
import LogInForm from '../Forms/LogInForm';
import SignUpForm from '../Forms/SignUpForm';
import { connect } from 'react-redux';

class App extends Component {
  render() {
    if (this.props.user !== {}) {
      return <Redirect to="/login" />
    }
    
    return (
      <section className="App">
        <Route exact path='/' render={() => <SearchForm />} />
        <Route exact path='/' render={() => <Container/>} />
        <Route exact path='/login' render={() => <LogInForm />} />
        <Route exact path='/signup' render={() => <SignUpForm />} />
      </section>
      );
    }
}

const mapStateToProps = ({ user }) => ({
  user
});

export default connect(mapStateToProps)(App);
