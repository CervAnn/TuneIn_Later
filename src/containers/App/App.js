import React, { Component } from "react";
import "./App.css";
import { Route, Link } from "react-router-dom";
import SearchForm from "../Forms/SearchForm/SearchForm";
import CardContainer from "../CardContainer/CardContainer";
import CardDetails from "../../Components/CardDetails/CardDetails";
import LogInForm from "../Forms/LogInForm/LogInForm";
import SignUpForm from "../Forms/SignUpForm/SignUpForm";
import { connect } from "react-redux";
import PropTypes from 'prop-types';

export class App extends Component {
  render() {
    return (
      <section className="App">
        <header>
          <Link to={"/"} className="header-link">
            <div className="headerContents">
              <h1 id="header-text">TuneIn Later</h1>
              <img id="headphones" src={require("./headphones.svg")} alt="" />
            </div>
          </Link>
          {!this.props.user ? (
            <article className="buttonContainer">
              <Link to="/login">
                <img
                  id="existingUser"
                  src={require("./user-silhouette.svg")}
                  alt=""
                />
                <button id="existingUser-button">LogIn</button>
              </Link>
              <Link to="/signup">
                <img id="newUser" src={require("./new-user.svg")} alt="" />
                <button id="newUser-button">SignUp</button>
              </Link>
            </article>
          ) : (
            <article className="buttonContainer">
              <Link to="/my-collection">
                <img id="viewFavorites" src={require("./vinyl.svg")} alt="" />
                <button id="viewFavorites-button">
                  View Favorites({this.props.favorites.length})
                </button>
              </Link>
              <Link to="/">
                <img id="logoutUser" src={require("./logout.svg")} alt="" />
                <button
                  id="logoutUser-button"
                  onClick="window.location.reload();"
                >
                  Logout
                </button>
              </Link>
            </article>
          )}
        </header>
        <Route exact path="/signup" render={() => <SignUpForm />} />
        <Route exact path="/" render={() => <SearchForm />} />
        <Route exact path="/login" render={() => <LogInForm />} />
        <Route
          exact
          path="/"
          render={() => <CardContainer displayType={"albums"} />}
        />
        <Route
          exact
          path="/my-collection"
          render={() => <CardContainer displayType={"favorites"} />}
        />
        <Route
          path="/albums/:id"
          render={({ match }) => {
            let foundAlbum = this.props.albums.find(album => {
              return album.collectionId == match.params.id;
            });
            return (
              <CardDetails
                key={foundAlbum.collectionId}
                artist_name={foundAlbum.artistName}
                album_name={foundAlbum.collectionName}
                artwork_url={foundAlbum.artworkUrl100}
                release_date={foundAlbum.releaseDate}
                primary_genre_name={foundAlbum.primaryGenreName}
                returnRoute={"/"}
              />
            );
          }}
        />
        <Route
          path="/favorites/:id"
          render={({ match }) => {
            let foundAlbum = this.props.favorites.find(favorite => {
              return favorite.album_id == match.params.id;
            });
            return (
              <CardDetails
                key={foundAlbum.album_id}
                artist_name={foundAlbum.artist_name}
                album_name={foundAlbum.collection_name}
                artwork_url={foundAlbum.artwork_url}
                release_date={foundAlbum.release_date}
                primary_genre_name={foundAlbum.primary_genre_name}
                returnRoute={"/my-collection"}
              />
            );
          }}
        />
      </section>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.user,
  albums: state.albums,
  favorites: state.favorites
});

export default connect(mapStateToProps)(App);

App.propTypes = {
  user: PropTypes.object,
  albums: PropTypes.array,
  favorites: PropTypes.array,
  mapStateToProps: PropTypes.func
}