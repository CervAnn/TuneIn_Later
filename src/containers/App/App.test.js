import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps } from './App';

describe('App', () => {
  let wrapper, mockUser, mockAlbums, mockFavorites;

  beforeEach(() => {
    mockUser = null;
    mockAlbums = [{ album_name: "Homecoming", artist_name: 'Beyonce', album_id: 12345 }]
    mockFavorites = [
      {
        album_id: 626204707,
        artist_name: 'Beyoncé',
        album_name: '4 (Expanded Edition)',
        price: 11.99,
        artwork_url: 'https://is2-ssl.mzstatic.com/image/thumb/Music6/v4/17/84/3a/17843a6d-1f2b-7e1e-a39f-3ff865110993/source/100x100bb.jpg',
        release_date: '2013-12-13T08:00:00Z',
        content_advisory_rating: 'Explicit',
        primary_genre_name: 'Pop',
      }
    ]

    wrapper = shallow(<App
      user={mockUser}
      albums={mockAlbums}
      favorites={mockFavorites}
    />);

  });

  it('should match the snapshot with all data passed in correctly', () => {

    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {

    it('should return an object with the current user', () => {
      const mockState = {
        albums: [{ album_name: "Homecoming", artist_name: 'Beyonce', album_id: 12345 }],
        user: { id: 1, name: 'Alan', email: 'alan@turing.io' },
        error: '',
        favorites: [
          {
            album_id: 626204707,
            artist_name: 'Beyoncé',
            album_name: '4 (Expanded Edition)',
            price: 11.99,
            artwork_url: 'https://is2-ssl.mzstatic.com/image/thumb/Music6/v4/17/84/3a/17843a6d-1f2b-7e1e-a39f-3ff865110993/source/100x100bb.jpg',
            release_date: '2013-12-13T08:00:00Z',
            content_advisory_rating: 'Explicit',
            primary_genre_name: 'Pop',
          }
        ]
      }
  
      const expectedUser = { id: 1, name: 'Alan', email: 'alan@turing.io' }
  
      expect(mapStateToProps(mockState).user).toEqual(expectedUser)
      expect(mapStateToProps(mockState).albums).toEqual(mockAlbums)
      expect(mapStateToProps(mockState).favorites).toEqual(mockFavorites)
    });

  });
})