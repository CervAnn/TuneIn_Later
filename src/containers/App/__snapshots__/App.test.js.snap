// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<section
  className="App"
>
  <header>
    <Link
      className="header-link"
      to="/"
    >
      <div
        className="headerContents"
      >
        <h1
          id="header-text"
        >
          TuneIn Later
        </h1>
        <img
          alt=""
          id="headphones"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "headphones.svg",
            }
          }
        />
      </div>
    </Link>
    <article
      className="buttonContainer"
    >
      <Link
        to="/login"
      >
        <img
          alt=""
          id="existingUser"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "user-silhouette.svg",
            }
          }
        />
        <button
          id="existingUser-button"
        >
          LogIn
        </button>
      </Link>
      <Link
        to="/signup"
      >
        <img
          alt=""
          id="newUser"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "new-user.svg",
            }
          }
        />
        <button
          id="newUser-button"
        >
          SignUp
        </button>
      </Link>
    </article>
  </header>
  <Route
    exact={true}
    path="/signup"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/my-collection"
    render={[Function]}
  />
  <Route
    path="/albums/:id"
    render={[Function]}
  />
  <Route
    path="/favorites/:id"
    render={[Function]}
  />
</section>
`;
